deepchem/data/data_loader.py

```diff
@@ -415,29 +384,119 @@ def _get_shards(self, input_files, shard_size):
       List of filenames to process
     shard_size: int
       The size of a shard of data to process at a time.
+
+    Returns
+    -------
+    Iterator over shards
     """
     return load_csv_files(input_files, shard_size)
 
-  def _featurize_shard(self, shard):
-    """Featurizes a shard of an input dataframe."""
-    return _featurize_smiles_df(
-        shard,
-        self.featurizer,
-        field=self.smiles_field,
-        log_every_n=self.log_every_n)
+  def _featurize_shard(self,
+                       shard: pd.DataFrame) -> Tuple[np.ndarray, np.ndarray]:
+    """Featurizes a shard of an input dataframe.
+
+    Parameters
+    ----------
+    shard: pd.DataFrame
+      DataFrame that holds a shard of the input CSV file
+
+    Returns
+    -------
+    features: np.ndarray
+      Features computed from CSV file.
+    valid_inds: np.ndarray
+      Indices of rows in source CSV with valid data.
+    """
+    logger.info("About to featurize shard.")
+    features = [elt for elt in self.featurizer(shard[self.feature_field])]
+    valid_inds = np.array(
+        [1 if np.array(elt).size > 0 else 0 for elt in features], dtype=bool)
+    features = [
+        elt for (is_valid, elt) in zip(valid_inds, features) if is_valid
+    ]
```

Comment:
From the view point of the performance, is it better to reduce the loop...? 
```python
features = []
valid_inds = []
for feat in self.featurizer(shard[self.feature_field])):
    is_valid = True if feat.size > 0 else False
    valid_inds.append(is_valid)
    if is_valid:
        features.append(feat)

```

Reply:
You can test it and see which is faster.  Based on my recent work, I think these lines probably have a negligible impact on performance, so it won't make a difference either way.

Reply:
In the interests of keeping the code succinct, I'm going to keep the current lines, but if it turns out to be a performance issue, will fix!